foundry/gui/player_lives.py

```diff
@@ -0,0 +1,528 @@
+from email.policy import default
+from PySide6.QtGui import QPixmap, Qt, QIntValidator
+from PySide6.QtWidgets import QBoxLayout, QLabel, QDialogButtonBox, QLineEdit, QGridLayout, QCheckBox, QGroupBox
+
+from dataclasses import dataclass
+from foundry.smb3parse.util.rom import Rom
+from foundry.game.File import ROM
+from foundry.gui.CustomDialog import CustomDialog
+from foundry.gui.HorizontalLine import HorizontalLine
+from foundry.smb3parse.util.code_edit_byte import CodeEditByte
+from foundry.smb3parse.util.code_edit_dict import CodeEditDict
+from foundry.core.redux_store import ReduxStore, Action
+
+@dataclass
+class _UIStrings:
+    """ All string values used in the UI. """
+
+    title = "Player Lives"
+    starting_lives = "Starting Lives"
+    continue_lives = "Continue Lives"
+    invalid_rom_warning = "The selected ROM has code modifications that are"  \
+        " incompatible with one or more features of this window. The affected" \
+        " features are visible but disabled."
+    death_takes_lives = "Subtract a life when the player dies"
+    title_1up = "Add a life when player gets 1up from:"
+    end_card_1up = "Level end card"
+    mushroom_1up = "1up Mushroom / Jumps on enemies "
+    dice_game_1up = "Dice Game"
+    roulette_1up = "Roulette Game"
+    card_game_1up = "Card Game"
+    hundred_coins_1up = "100 Coins"
+
+@dataclass
+class ActionNames:
+    """ All user actions that can be performed. """
+
+    load = "[PlayerLives] Load"
+    starting_lives = "[PlayerLives] StartingLives"
+    continue_lives = "[PlayerLives] ContinueLives"
+    death_takes_lives = "[PlayerLives] DeathTakesLives"
+    hundred_coins_1up = "[PlayerLives] 100Coins"
+    end_card_1up = "[PlayerLives] EndCard"
+    mushroom_1up = "[PlayerLives] Mushroom"
+    dice_game_1up = "[PlayerLives] DiceGame"
+    roulette_1up = "[PlayerLives] Roulette"
+    card_game_1up = "[PlayerLives] CardGame"
+
+@dataclass
+class State:
```

Comment:
A docstring describing the elements of this class would be greatly useful.  Some of these may be limited or obscured in some fashion.  Creating """Attributes...""" would allow for any of this critical information to be cited in a centralized location.